src/main/java/co/cask/spanner/sink/SpannerSink.java

```diff
@@ -0,0 +1,104 @@
+import co.cask.cdap.etl.api.batch.BatchRuntimeContext;
+import co.cask.cdap.etl.api.batch.BatchSinkContext;
+import co.cask.gcs.GCPUtil;
+import co.cask.hydrator.common.ReferenceBatchSink;
+import co.cask.hydrator.common.batch.sink.SinkOutputFormatProvider;
+import co.cask.spanner.SpannerConstants;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+
+
+/**
+ * This class extends {@link ReferenceBatchSink} to write to Google Cloud Spanner.
+ *
+ * Uses a {@link SpannerOutputFormat} and {@link SpannerOutputFormat.SpannerRecordWriter} to configure
+ * and write to spanner. The <code>prepareRun</code> method configures the job by extracting
+ * the user provided configuration and preparing it to be passed to {@link SpannerOutputFormat}.
+ *
+ */
+@Plugin(type = "batchsink")
+@Name(SpannerSink.NAME)
+@Description("Batch sink to write to Cloud Spanner. Cloud Spanner is a fully managed, mission-critical, " +
+  "relational database service that offers transactional consistency at global scale, schemas, " +
+  "SQL (ANSI 2011 with extensions), and automatic, synchronous replication for high availability.")
+public final class SpannerSink
+  extends ReferenceBatchSink<StructuredRecord, NullWritable, StructuredRecord> {
+  public static final String NAME = "Spanner";
+  private final SpannerSinkConfig config;
+
+  /**
+   * Initializes <code>SpannerSink</code>.
+   * @param config
+   */
+  public SpannerSink(SpannerSinkConfig config) {
+    super(config);
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate();
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    config.validate();
+    Configuration configuration = new Configuration();
+    String projectId = GCPUtil.getProjectId(config.project);
+    configuration.set(SpannerConstants.PROJECT_ID, projectId);
+    if (config.serviceFilePath != null) {
+      configuration.set(SpannerConstants.SERVICE_ACCOUNT_FILE_PATH, config.serviceFilePath);
+    }
+    configuration.set(SpannerConstants.INSTANCE_ID, config.instance);
+    configuration.set(SpannerConstants.DATABASE, config.database);
+    configuration.set(SpannerConstants.TABLE_NAME, config.table);
+    configuration.set(SpannerConstants.SPANNER_WRITE_BATCH_SIZE, String.valueOf(config.getBatchSize()));
+    configuration.set(SpannerConstants.SCHEMA, config.getSchema().toString());
+    context.addOutput(Output.of(config.referenceName,
+                                new SinkOutputFormatProvider(SpannerOutputFormat.class, configuration)));
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    super.initialize(context);
+    config.validate();
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<KeyValue<NullWritable,
```

Comment:
nit: better to keep the variable all on one line, let's move Emitter down to the next line